tests/scenario/test_charm.py

```diff
@@ -0,0 +1,601 @@
+#! /usr/bin/env python
```

Comment:
On the whole I think these Scenario tests are really clean at a glance (and don't overuse fixtures). I think this file would make a good test to go over with a fine-toothed comb to see what APIs we want to clean up in Scenario before going official -- it really makes the awkward APIs stand out.

Update: looks like you've somewhat been doing this as you go with SCENARIO-NOTE -- thanks.

Reply:
FYI, I've since copied over the `SCENARIO-NOTE` notes from here and the Charms where I converted Harness tests to Scenario over to the main Google doc so that it's all in one place.